erica/api/v2/endpoints/est.py

```diff
@@ -22,15 +20,11 @@ async def send_est(est_data_client_identifier: TaxDeclarationDto):
     Route for sending a tax declaration using the job queue.
     :param est_data_client_identifier: payload with client identifier and the JSON input data for the tax declaration.
     """
-    try:
-        result = get_job_service(RequestType.send_est).add_to_queue(
-            est_data_client_identifier.payload, est_data_client_identifier.clientIdentifier,
-            RequestType.send_est)
-        return RedirectResponse(url='ests/' + str(result.request_id), status_code=201)
-    # TODO specific exception and correct mapping to JSON error response?
-    except Exception:
-        logging.getLogger().info("Could not send est", exc_info=True)
-        return JSONResponse(status_code=422, content=generate_error_response())
+
+    result = get_job_service(RequestType.send_est).add_to_queue(
```

Comment:
We still have in every API use case the responses model:
- For posting a job we have both 422 and 500 with the errorCode/errorMessage JSON
- For getting a job we have both 404 and 500 with the errorCode/errorMessage JSON

The exception handlers you wrote are only meant for the "getting a job" methods so:
- When posting a job unexpected errors would always have the default starlette format right? So no errorCode/errorMessage
- When getting a job the 404 error would have the expected errorCode/errorMessage format but unexpected errors would not fulfill the expected body of the response model.

This means we should:
- Remove the responses model when posting a job (422 and 500) right?
- Remove the 500 response model when getting a job right?

Reply:
> The exception handlers you wrote are only meant for the "getting a job" methods so:
No, they are for the whole application. So, whenever the application does not handle an EntityNotFoundError the exception handler is called. The same is true for any Exception (the internal server error comes in)

422 would still occur through the fastApi

Reply:
Therefore, I am not sure if we need to adapt something?

Reply:
Yes I meant you set that exception handlers for the get-jobs because our EntityNotFoundError can, as the code is right now, only happen in those use cases.
We were using before, as specified in the tech spec, the 422 for the case we had an expected error when posting the job. But now only the EntityNotFoundError (404) and the other one for the wrong request type (default 307) will be handled by your exception handler.
The responses models of the API methods are just definitions about how the responses look like when certain HTTP codes are returned. But for those models to be fulfilled, we have to manually generate the JSON response (meaning adding errorCode and errorMessage)

422 can still ocurrs yes but will have the default FastAPI json response

Reply:
Okay, then I understood you correctly: you would say that we should update those definitions to have the correct api description, right?

- get rid of 404 -> use 307 instead
- only have that for the gets
- get rid of the custom 422 -> we could also overwrite the exception handler for the validation error if we wanted to. But I would say that it is probably better to stay with the fastapi implementation for that one.

Thanks for pointing that out and having an eye for that detail👍🏼

Reply:
I think we have some bad misunderstanding here. Our current definitions of the response models are:
### POST Job
- For **_HTTP 422_** --> ErrorRequestQueue (meaning errorCode and errorMessage for the JSON response)
- For **_HTTP 500_** --> ErrorRequestQueue
- For the successful response with 201 we have the RedirectResponse object (no problem here)

### GET Job
- For HTTP 200 --> we have the specific respones DTO for every use case, so no problem here.
- For HTTP 404 --> ErrorRequestQueue (this one would be ok with your handler, since we are creating the JSON with the errorCode and errorMessage)
- For **_HTTP 500_** --> ErrorRequestQueue

As FastAPI states in their documentation [here](https://fastapi.tiangolo.com/advanced/additional-responses/): _But for those additional responses you have to make sure you return a Response like JSONResponse directly, with your status code and content._

This means, we are defining an specific response model (errorCode/errorMessage) that we are not fulfilling, since FastAPI will be automatically handling those HTTP errors (the ones marked bold and italic above)

I haven't actually tested what happens if and error is thrown of which we defined a response model but for which we didn't manually create our defined response with errorCode/errorMessage.

So my proposal is simple:
- Remove those response models, for the HTTP codes we are not handling ourselves (422 and 500 for POST and 500 for GET)
or
- Adding another exception handler for at least the 500, but I think you wanted to avoid robbing fast api from catching internal server errors right?